api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsService.java

```diff
@@ -341,19 +341,34 @@ public void extendInitialCreditsExpiration(DbUser user) {
     DbUserInitialCreditsExpiration userInitialCreditsExpiration =
         user.getUserInitialCreditsExpiration();
     if (userInitialCreditsExpiration == null) {
-      throw new WorkbenchException("User does not have initial credits expiration set.");
+      throw new WorkbenchException(
+          "User does not have initial credits expiration set, so they cannot extend their expiration date.");
+    }
+
+    if (institutionService.shouldBypassForCreditsExpiration(user)) {
+      throw new WorkbenchException(
```

Comment:
will this result api to return 500 or 400? I think we should return 400 in this case

Reply:
Good call. It was 500. I changed it to 400.